WPMedia/fusion-news-theme-blocks
# Contact info component and integration with checkout

## Description
Creation of the contact info component and integration with checkout.  One thing of note: I moved the `FormInputField` component over from Identity to Shared Styles and also added a select type.

## Jira Ticket
[- [TMEDIA-](https://arcpublishing.atlassian.net/browse/TMEDIA-)](https://arcpublishing.atlassian.net/browse/TMEDIA-536)

## Acceptance Criteria
Have an Email Address field with associated label
Display a blank e-mail address field if the user is not logged in
Email address field has e-mail address validation (same as on Sign Up block)
Show the user’s email address with a Log Out link if logged in via email
Show the user’s email address with a Log Out link if logged in via Social account
Have a First Name input field with associated label
Field is a required field
Display a blank field if the user is not logged in
Displays the users First Name if the firstName field from their profile is available
Have a Last Name input field with associated label
Field is a required field
Display a blank field if the user is not logged in
Displays the users Last Name if the lastName field from their profile is available
Have a Country Drop Down Selection field with associated label
Field is a required field
Always display field as empty and make user make selection
Component will take a function as a prop that is used to pass the above fields (Email, FirstName, LastName, Country) back to the parent component
Text strings should be translated where applicable

## Test Steps
- Add test steps a reviewer must complete to test this PR

1. Checkout this branch `git checkout TMEDIA-536-contact-info`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/shared-styles,@wpmedia/subscriptions-block,@wpmedia/identity-block,@wpmedia/header-nav-chain-block,@wpmedia/site-hierarchy-content-block
`
3. Create a checkout page.  Add the checkout block to the page. You should then see this: 
![image](https://user-images.githubusercontent.com/833461/145117972-5ce4af6d-9b01-44b7-a10f-1826739822a4.png)
4. Go to the published homepage on your local site for the Gazzette and login with Google.  Then go to the published page for your local checkout and refresh and this is what you should now see: 
![image](https://user-images.githubusercontent.com/833461/145118126-56f8dc13-c0ad-48e8-b675-3ac3ae3787f7.png)
5. Click the sign out link and it verify it signs you out (can confirm by refreshing our published homepage in a seperate tab) and that it replaces the text with an email input control.

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [x ] Confirmed all the test steps a reviewer will follow above are working.
- [ x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [x ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [x ] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x ] Confirmed this PR has unit test files
  - [x ] Ran `npm run test`, made sure all tests are passing
please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._
